### PR TITLE
Feature/update sign request data

### PR DESCRIPTION
**Pull request by mikonieminen**
Update the data format that is expected by `cosigner/sign` path.

### CODE REVIEW

File: src/message_builder.rs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        DeepHashChunk::Chunk(BUNDLR_AS_BUFFER.into()),
+        DeepHashChunk::Chunk(ONE_AS_BUFFER.into()),
+        DeepHashChunk::Chunk(tx.as_bytes().to_owned().into()),
+        DeepHashChunk::Chunk(size.to_string().as_bytes().to_owned().into()),
+        DeepHashChunk::Chunk(fee.to_string().as_bytes().to_owned().into()),
+        DeepHashChunk::Chunk(currency.as_bytes().to_owned().into()),
+        DeepHashChunk::Chunk(block.to_string().as_bytes().to_owned().into()),
+        DeepHashChunk::Chunk(validator.as_bytes().to_owned().into()),
+    ]))
+    .unwrap();
```
Comment (mikonieminen) [inline src/message_builder.rs]: @joshbenaron can you review and verify that his is correct data format send by bundler node to validator node? Then in `src/server/routes/sign.rs` this message is received, verified and signed with validator key.

File: src/server/routes/sign.rs [hunk 2]
```diff
@@ -69,18 +79,16 @@ impl UnsignedBody {
...
     {
-        let tx_id = self.id.as_bytes().to_vec();
         let bundler_address = key_manager.bundler_address().to_string();
 
         let signature_data = deep_hash(DeepHashChunk::Chunks(vec![
             DeepHashChunk::Chunk(VALIDATOR_AS_BUFFER.into()),
             DeepHashChunk::Chunk(ONE_AS_BUFFER.into()),
-            DeepHashChunk::Chunk(tx_id.into()),
+            DeepHashChunk::Chunk(self.id.as_bytes().to_owned().into()),
             DeepHashChunk::Chunk(bundler_address.into()),
```
Comment (mikonieminen) [inline src/server/routes/sign.rs]: @joshbenaron is this correct data for validator signature of the received data or should we include other fields too?

File: src/message_builder.rs (hunk 1 above)
Comment (joshbenaron) [reply src/message_builder.rs]: LGTM

File: src/server/routes/sign.rs [hunk 3]
```diff
@@ -66,19 +79,22 @@ impl UnsignedBody {
...
-            DeepHashChunk::Chunk(tx_id.into()),
-            DeepHashChunk::Chunk(bundler_address.into()),
+            DeepHashChunk::Chunk(self.id.as_bytes().to_owned().into()),
+            DeepHashChunk::Chunk(self.size.to_string().as_bytes().to_owned().into()),
+            DeepHashChunk::Chunk(self.fee.to_string().as_bytes().to_owned().into()),
+            DeepHashChunk::Chunk(self.currency.as_bytes().to_owned().into()),
+            DeepHashChunk::Chunk(self.block.to_string().as_bytes().to_owned().into()),
+            DeepHashChunk::Chunk(self.validator.as_bytes().to_owned().into()),
+            DeepHashChunk::Chunk(bundler_address.as_bytes().to_owned().into()),
         ]))
```
Comment (mikonieminen) [inline src/server/routes/sign.rs:98]: @joshbenaron does this now look correct? Now the validator signature includes also all fields plus adds the `bundler_address`.

Comment (joshbenaron) [reply src/server/routes/sign.rs:98]: LGTM

**Pull request merged**